src/main/capi/scalar_function-c.cpp

```diff
@@ -45,6 +45,11 @@ duckdb::ScalarFunction &GetCScalarFunction(duckdb_scalar_function function) {
 unique_ptr<FunctionData> BindCAPIScalarFunction(ClientContext &context, ScalarFunction &bound_function,
                                                 vector<unique_ptr<Expression>> &arguments) {
 	auto &info = bound_function.function_info->Cast<CScalarFunctionInfo>();
+	if (bound_function.HasVarArgs()) {
+		for (const auto &expr : arguments) {
+			bound_function.arguments.push_back(expr->return_type);
```

Comment:
Could you expand on the code here? `varargs` explicitly has a type (i.e. it is not just true or false) - it looks to me that in this manner that type is ignored and is essentially set to `LogicalType::ANY`.